### ISSUE TITLE
heroku_inbound_ruleset using counts isn't possible

### ISSUE THREAD

**Issue by talbright** [terraform-providers/terraform-provider-heroku#117]
```sh
terraform -v
Terraform v0.11.8
+ provider.heroku v1.3.0
```

### Affected Resource(s)
- heroku_space_inbound_ruleset

### Terraform Configuration Files
```hcl
locals {
  inbound_rules = ["61.120.150.0/28", "62.17.146.0/26"]
}

resource "heroku_space" "test_talbright" {
  name         = "test-talbright"
  organization = "${var.organization}"
  region       = "${var.region}"
  shield       = "false"
}

resource "heroku_space_inbound_ruleset" "ruleset" {
  count = "${length(local.inbound_rules)}"
  space = "test-talbright"

  rule {
    action = "allow"
    source = "${element(local.inbound_rules,count.index)}"
  }
}
```

### Output
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + heroku_space_inbound_ruleset.ruleset[0]
      id:                     <computed>
      rule.#:                 "1"
      rule.1473555236.action: "allow"
      rule.1473555236.source: "61.120.150.128/28"
      space:                  "test-talbright"

  + heroku_space_inbound_ruleset.ruleset[1]
      id:                     <computed>
      rule.#:                 "1"
      rule.2428952940.action: "allow"
      rule.2428952940.source: "62.17.146.128/26"
      space:                  "test-talbright"


Plan: 2 to add, 0 to change, 0 to destroy.
```

### Expected Behavior
`heroku_space_inbound_ruleset` rules can't be generated by using a count with a list. This means every single rule has to be hardcoded 😢 

Perhaps the resource should either be modeled closer to https://www.terraform.io/docs/providers/aws/r/security_group_rule.html, or a new resource should be created to encapsulate the same approach. 

### Actual Behavior
`heroku_space_inbound_ruleset` with count only one rule is generated, the last item in the list. This is because each API call to `PUT /spaces/{space-name}/inbound-ruleset` replaces/overwrites the ruleset.

**Comment by joestump** [comment]
Having this resource replicate the `aws_security_group_rule` would require the API endpoint to be rewritten by our friends at Heroku because as you mention the API overwrites/replaces rulesets. 

The good news is that you _can_ pass a list to HCL blocks. This is some example code I came up with for another person looking for similar behavior: 

```hcl
provider "heroku" {}

resource "heroku_space" "foobar" {
  name         = "joe-testing-1"
  organization = "my-team"
  region       = "virginia"
}

locals {
  prd = [
    {
      action = "allow"
      source = "8.8.8.8/32"
    }
  ]

  vpn = [
    {
      action = "allow"
      source = "8.8.4.4/32"
    }
  ]

  space = [
    {
      action = "allow"
      source = "${heroku_space.foobar.outbound_ips[0]}/32"
    },
    {
      action = "allow"
      source = "${heroku_space.foobar.outbound_ips[1]}/32"
    },
    {
      action = "allow"
      source = "${heroku_space.foobar.outbound_ips[2]}/32"
    },
    {
      action = "allow"
      source = "${heroku_space.foobar.outbound_ips[3]}/32"
    }
  ]
}

resource "heroku_space_inbound_ruleset" "lock_space_to_itself" {
  space = "${heroku_space.foobar.id}"
  rule = ["${local.prd}"]
  rule = ["${local.vpn}"]
  rule = ["${local.space}"]
}
```

**Comment by joestump** [comment]
TL;DR you can pass lists of maps into an HCL block and it'll just work.

**Comment by talbright** [comment]
Helpful suggestion! I'll try this out. There's an additional axis here, that is my space has a count so that introduces splat syntax.

It's not strictly true that support for this would require an API change. One would have to get the rules via the api, and do some comparisons, then send the set back.

**Comment by talbright** [comment]
This didn't end up working for me, so I just used the older deprecated `trusted_ip_ranges`
